src/resolve-config.js

```diff
@@ -12,10 +12,20 @@ const syncNoCache = cosmiconfig("prettier", { cache: false, sync: true });
 function editorConfigToPrettier(filePath) {
   const editorConfig = editorconfig.parseSync(filePath);
   const result = {};
-  result.useTabs = editorConfig.indent_style === "tab";
-  result.tabWidth =
-    editorConfig.indent_size || editorConfig.tab_width || result.tabWidth;
-  result.printWidth = editorConfig.max_line_length || result.printWidth;
+
+  if (editorConfig.indent_style) {
+    result.useTabs = editorConfig.indent_style === "tab";
+  }
+
+  const tabWidth = editorConfig.indent_size || editorConfig.tab_width;
+  if (tabWidth) {
+    result.tabWidth = tabWidth;
+  }
+
+  if (editorConfig.max_line_length) {
+    result.printWidth = editorConfig.max_line_length;
+  }
+
```

Comment:
Is it important that the `.useTabs`, `.tabWidth` and `.printWidth` properties of `result` are missing rather than present with the value `undefined` if the corresponding EditorConfig setting is missing?

Couldn't we just do:

```js
const editorConfig = editorconfig.parseSync(filePath);
return {
  useTabs: ...,
  tabWidth: ...,
  printWidth: ...
}
```

While being an odd configuration, isn't it valid to set `indent_size`, `tab_width` and `max_line_length` to `0`, and as such we should respect that?

Reply:
> Is it important that the .useTabs, .tabWidth and .printWidth properties of result are missing rather than present with the value undefined if the corresponding EditorConfig setting is missing?

Yes, I think that's what fixed this test: https://travis-ci.org/prettier/prettier/jobs/272596635#L379

> While being an odd configuration, isn't it valid to set indent_size, tab_width and max_line_length to 0, and as such we should respect that? Note that 0 if a falsy value.

Hmm, it looks like [tab_width](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#tab_width) and [max_line_length](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#max_line_length) must be positive. I suppose someone could set `indent_size` to 0, but I'm not sure if that edge case is worth supporting right now. At least, I haven't had time to put effort into it.

Reply:
I guess it's all fine then 👍